zeppelin-plugins/launcher/k8s-standard/src/main/java/org/apache/zeppelin/interpreter/launcher/K8sRemoteInterpreterProcess.java

```diff
@@ -319,9 +321,25 @@ Properties getTemplateBindings(String userName) {
     if (isSpark()) {
       int webUiPort = 4040;
       k8sProperties.put("zeppelin.k8s.spark.container.image", sparkImage);
+
+      // There is already initial value following --driver-java-options added in interpreter.sh
+      // so we need to pass spark.driver.defaultJavaOptions and spark.driver.extraJavaOptions
+      // as SPARK_DRIVER_EXTRAJAVAOPTIONS_CONF env variable to build spark-submit command correctly.
+      StringBuilder driverExtraJavaOpts = new StringBuilder();
+      if (properties.containsKey(SPARK_DRIVER_DEFAULTJAVAOPTS)) {
+        driverExtraJavaOpts.append((String) properties.remove(SPARK_DRIVER_DEFAULTJAVAOPTS));
+      }
+      if (properties.containsKey(SPARK_DRIVER_EXTRAJAVAOPTS)) {
+        driverExtraJavaOpts.append(" ");
```

Comment:
nit. You can use `StringJoiner(" ")` to reduce `.append(" ");`

Reply:
fixed